connectors/config.py

```diff
@@ -59,8 +59,9 @@ def _default_config():
                 "max_concurrency": 5,
                 "chunk_max_mem_size": 5,
                 "concurrent_downloads": 10,
-                "max_retries": 3,
```

Comment:
I technically moved this `elasticsearch.bulk.max_retries` to the new `elasticsearch.max_retries` + increased it. So it won't work _exactly_ like before, rather retry harder.

I don't think we've done a lot of it before, should I write a warning that this setting is deprecated and the new one should be used?

Reply:
Yeah, adding more information about this change would be useful. A sentence or two in the release notes would be great too.

Reply:
I've added info to the release notes in the PR + made a change in the commit https://github.com/elastic/connectors/pull/2063/commits/a4d4f8bd1b15113d12f119920ea5ecb0753b48a7 to check deprecated configs and report to terminal.

Example message in the terminal:

```
artemshelkovnikov-m2@Work-M2 connectors-py % make run
bin/elastic-ingest
[FMWK][14:09:29][INFO] Running connector service version 8.13.0.0
[FMWK][14:09:29][INFO] Loading config from /Users/artemshelkovnikov-m2/git_tree/connectors-py/connectors/../config.yml
[FMWK][14:09:29][WARNING] Configuration option 'elasticsearch.bulk.max_retries' is deprecated: Use elasticsearch.max_retries instead.
```

Reply:
We currently don't have any special handling for bulk tbh, before it was just blindly retrying any error `{bulk.max_retries}` times. Now I introduced this logic in a slightly better manner to all ES calls, so I don't think we want it separate.

Re: retry only failed operations - we don't do this at all now, and this will be addressed later.

Reply:
>We currently don't have any special handling for bulk tbh

I know. My point is, there will be special retry logic for `bulk` function, and this `max_retries` is originally for `bulk` function, so if you keep this `max_retries` unchanged (reserved for future `bulk` retry handling), and introduce new configs in `elasticsearch` level for all non-bulk functions, you can avoid breaking change and deprecation warning.

Just a suggestion, I'm not so worried about config deprecation.

Reply:
Good call, I'll do it like this while we're waiting for response about our config deprecation strategy!

Reply:
What I've done now:

Increased default `elasticsearch.bulk.max_retries` to 5 + added an option `elasticsearch.bulk.retry_interval` set to 10 so that current behaviour of bulk insert would be absolutely same as other calls to Elasticsearch that are configured by `elasticsearch.max_retries` and `elasticsearch.retry_interval`.

We can deprecate `elasticsearch.bulk.max_retries|retry_interval` later, what's important to me now is that these two by default would be same.